Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2682,8 +2730,114 @@ namespace OpenInfraPlatform
 					const typename IfcEntityTypesT::IfcNonNegativeLengthMeasure & parameter) const noexcept(false)
 				{
 					throw oip::UnhandledException(curve);
+
 				}
 
+				template <typename TCurve>
+				carve::geom::vector<3> getDirectionOfCurve(const EXPRESSReference<TCurve>& curve,
+					const double& parameter) const noexcept(false)
+				{
+					throw oip::UnhandledException(curve);
+
+				}
+
+				/**********************************************************************************************/
+				/*! \brief Converts \c IfcCartesianPoint to a angle parameter and passes it to getDirectionOfCurve.
+				* \param[in] circle					A pointer to data from \c IfcCircle.
+				* \param[in] point					A pointer to data from \c IfcCartesianPoint.
+				* \return							The Angle in radians.
+				*/
+				template<>
+				carve::geom::vector<3> getDirectionOfCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
+					const typename IfcEntityTypesT::IfcCartesianPoint& cartesianPoint) const noexcept(false)
+				{
+					carve::geom::vector<3> point = placementConverter->convertIfcCartesianPoint(cartesianPoint);
+					carve::math::Matrix placement = placementConverter->convertIfcAxis2Placement(circle->Position);
+					carve::math::Matrix inversePlacement = GeomUtils::computeInverse(placement);
+					carve::geom::vector<3> directionFromCenterToPoint = inversePlacement * point - carve::geom::VECTOR(0., 0., 0.);
+					// if the radial vector from Center to a point has coordinates (a,b), then the direction vector at that point is (−b,a)
+					return carve::geom::VECTOR(-directionFromCenterToPoint.y, directionFromCenterToPoint.x, 0.)
+				}
+
+				/*! \brief Converts \c IfcParameterValue to a angle parameter and passes it to getDirectionOfCurve.
+				* \param[in] circle					A pointer to data from \c IfcCircle.
+				* \param[in] parameter				A pointer to data from \c IfcParameterValue.
+				* \return							The Angle in radians.

```

Comment:
it doesn't return the angle, but rather the direction of the tangent. This should be addressed in all doxy-comments (I wasn't paying attention above ...)

Reply:
Resolved: 4192532